scripts/testnet.sh

```diff
@@ -38,94 +38,47 @@ get_rpc() {
   local environment_id="${1}"
   local network_id=$(cat "${mydir}/../packages/core/protocol-config.json" | jq -r ".environments.\"${environment_id}\".network_id")
   local unresolved_rpc=$(cat "${mydir}/../packages/core/protocol-config.json" | jq -r ".networks.\"${network_id}\".default_provider")
-  echo "${unresolved_rpc}" | envsubst
-}
-
-# $1=account (hex)
-# $2=rpc
-wallet_balance() {
-  local address=${1}
-  local rpc=${2}
-
-  yarn run --silent ethers eval "new ethers.providers.JsonRpcProvider('${rpc}').getBalance('${address}').then(b => formatEther(b))"
-}
-
-# $1=rpc
-funding_wallet_address() {
-  local rpc=${1}
 
-  # the value of FUNDING_PRIV_KEY must be prefixed with 0x
-  yarn run --silent ethers --rpc "${rpc}" --account ${FUNDING_PRIV_KEY} eval 'accounts[0].getAddress().then(a => a)'
+  echo "${unresolved_rpc}" | envsubst
 }
 
 # $1=account (hex)
-# $2=environment_id
-# $3=optional: hopr token contract
+# $2=environment
 fund_if_empty() {
   local address="${1}"
-  local environment_id="${2}"
-  local token_contract="${3:-}"
-
-  local funding_address funding_balance balance
-
-  local rpc=$(get_rpc "${environment_id}")
+  local environment="${2}"
 
-  funding_address=$(funding_wallet_address "${rpc}")
+  # start funding in parallel
+  PRIVATE_KEY="${FUNDING_PRIV_KEY}" ${mydir}/fund-address.ts \
+	  --environment ${environment} --address ${address} --target ${min_funds} &
 
-  log "Checking balance of funding wallet ${funding_address} using RPC ${rpc}"
-  funding_balance="$(wallet_balance "${funding_address}" "${rpc}")"
+  PRIVATE_KEY="${FUNDING_PRIV_KEY}" ${mydir}/fund-address.ts \
```

Comment:
Not sure I understand this line fully:

1. You're calling a `.ts` script which is highly system-specific. Isn't it better to explicitly call tsnode?
2. You're running something in the background (`&` sign in the end) and assign the return result to a `PRIVATE_KEY`. What is that value?

Reply:
No. `PRIVATE_KEY` is set so that it is accessible to the script's execution environment. The shell will know how to execute the script based on its shebang.

Reply:
Yeah, right. Okay then.